lib/tape.js

```diff
@@ -0,0 +1,259 @@
+				});
+			},
+
+			function testOptionArgument() {
+				// Convert Tape option parameters, test([name], [opts], cb)
+				ast.find(j.CallExpression, {
+					callee: {name: testFunctionName}
+				}).forEach(p => {
+					p.value.arguments.forEach(a => {
+						if (a.type === 'ObjectExpression') {
+							a.properties.forEach(tapeOption => {
+								const tapeOptionKey = tapeOption.key.name;
+								const tapeOptionValue = tapeOption.value.value;
+								if (tapeOptionKey === 'skip' && tapeOptionValue === true) {
+									p.value.callee.name += '.cb.serial.skip';
+								}
+
+								if (tapeOptionKey === 'timeout') {
+									throw new Error('Codemod transformation of "timeout" option is not supported');
+								}
+							});
+
+							p.value.arguments = p.value.arguments.filter(pa => pa.type !== 'ObjectExpression');
+						}
+					});
+				});
+			},
+
+			function updateTapeComments() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: {name: 'comment'}
+					}
+				})
+				.forEach(p => {
+					p.node.callee = 'console.log';
+				});
+			},
+
+			function updateThrows() {
+				// The semantics of t.throws(fn, expected, msg) is different.
+				// - Tape: if `expected` is a string, it is set to msg
+				// - AVA: if `expected` is a string it is transformed to a function
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: {name: 'throws'}
+					},
+					arguments: arg => arg.length === 2 && arg[1].type === 'Literal' && typeof arg[1].value === 'string'
+				})
+				.forEach(p => {
+					const [fn, msg] = p.node.arguments;
+					p.node.arguments = [fn, j.literal(null), msg];
+				});
+			},
+
+			function updateTapeOnFinish() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: testFunctionName},
+						property: {name: 'onFinish'}
+					}
+				})
+				.forEach(p => {
+					p.node.callee.property.name = 'after.always';
+				});
+			},
+
+			function rewriteTestCallExpression() {
+				// To be on the safe side we rewrite the test(...) function to
+				// either test.cb.serial(...) or test.serial(...)
+				//
+				// - .serial as Tape runs all tests serially
+				// - .cb for tests containing t.end, as we cannot detect if the
+				//   test have any asynchronicity.
+				ast.find(j.CallExpression, {
+					callee: {name: 'test'}
+				}).forEach(p => {
+					// TODO: if t.end is in the scope of the test function we could
+					// remove it and not use cb style.
+					const containsEndFunction = j(p).find(j.CallExpression, {
+						callee: {
+							object: {name: 't'},
+							property: {name: 'end'}
+						}
+					}).size() > 0;
+
+					const newTestFunction = containsEndFunction ? 'cb.serial' : 'serial';
+
+					p.node.callee = j.memberExpression(
+						j.identifier('test'),
+						j.identifier(newTestFunction)
+					);
+				});
+			}
+		];
+
+		transforms.forEach(t => t());
+	}
+
+	return ast.toSource({quote: 'single'});
```

Comment:
So `recast` doesn't seem to infer anything around quoting. It will keep whatever the parser added, but when you do new literals it doesn't know how to quote that... Pretty annoying. I could find a workaround, so I just detect it. See https://github.com/avajs/ava-codemods/pull/28/commits/f05b6da71b767c929d84cbb5da28b40132a81318

Might be something to raise in https://github.com/facebook/jscodeshift